gui/webdisplay/src/RWebWindowsManager.cxx

```diff
@@ -517,28 +572,56 @@ void RWebWindowsManager::Unregister(RWebWindow &win)
 //////////////////////////////////////////////////////////////////////////
 /// Provide URL address to access specified window from inside or from remote
 
-std::string RWebWindowsManager::GetUrl(const RWebWindow &win, bool remote)
+std::string RWebWindowsManager::GetUrl(RWebWindow &win, bool remote, std::string *produced_key)
 {
    if (!fServer) {
       R__LOG_ERROR(WebGUILog()) << "Server instance not exists when requesting window URL";
       return "";
    }
 
    std::string addr = "/";
-
    addr.append(win.fWSHandler->GetName());
-
    addr.append("/");
 
+   bool qmark = false;
+
+   std::string key;
+
+   if (win.IsRequireAuthKey() || produced_key) {
+      key = win.GenerateKey();
+      if (key.empty())
+         return "";
```

Comment:
Should this rather throw or assert?

Reply:
Error message will be produced in `win.GenerateKey()` call. One can put assert there if we decide to do this.

Reply:
Since we _know_ that `GenerateKey()` won't come back with an empty key, I think it is the right place to have an `assert()` here. It would ensure that post condition and document it at the same time.